independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/ContextInstancesGenerator.java

```diff
@@ -0,0 +1,263 @@
+        // try {
+        //    Set<ContextInstanceHandle<?>> ret = new HashSet<>();
+        //    ContextInstanceHandle<?> copy = this.1;
+        //    if (copy != null) {
+        //       ret.add(copy);
+        //    }
+        //    return ret;
+        // } catch(Throwable t) {
+        //    lock.unlock();
+        //    throw t;
+        // }
+        ResultHandle lock = getAllPresent.readInstanceField(lockField, getAllPresent.getThis());
+        getAllPresent.invokeInterfaceMethod(MethodDescriptors.LOCK_LOCK, lock);
+        TryBlock tryBlock = getAllPresent.tryBlock();
+        ResultHandle ret = tryBlock.newInstance(MethodDescriptor.ofConstructor(HashSet.class));
+        for (FieldDescriptor field : idToField.values()) {
+            ResultHandle copy = tryBlock.readInstanceField(field, tryBlock.getThis());
+            tryBlock.ifNotNull(copy).trueBranch().invokeInterfaceMethod(MethodDescriptors.SET_ADD, ret, copy);
+        }
+        tryBlock.invokeInterfaceMethod(MethodDescriptors.LOCK_UNLOCK, lock);
+        tryBlock.returnValue(ret);
+        CatchBlockCreator catchBlock = tryBlock.addCatch(Throwable.class);
+        catchBlock.invokeInterfaceMethod(MethodDescriptors.LOCK_UNLOCK, lock);
+        catchBlock.throwException(catchBlock.getCaughtException());
+    }
+
+    private void implementClear(ClassCreator applicationContextInstances, Map<String, FieldDescriptor> idToField,
+            FieldDescriptor lockField) {
+        MethodCreator clear = applicationContextInstances.getMethodCreator("clear", void.class).setModifiers(ACC_PUBLIC);
+        // lock.lock();
+        // try {
+        //    this.1 = null;
+        //    lock.unlock();
+        // } catch(Throwable t) {
+        //    lock.unlock();
+        //    throw t;
+        // }
+        ResultHandle lock = clear.readInstanceField(lockField, clear.getThis());
+        clear.invokeInterfaceMethod(MethodDescriptors.LOCK_LOCK, lock);
+        TryBlock tryBlock = clear.tryBlock();
+        for (FieldDescriptor field : idToField.values()) {
+            tryBlock.writeInstanceField(field, tryBlock.getThis(), tryBlock.loadNull());
+        }
+        tryBlock.invokeInterfaceMethod(MethodDescriptors.LOCK_UNLOCK, lock);
+        tryBlock.returnVoid();
+        CatchBlockCreator catchBlock = tryBlock.addCatch(Throwable.class);
+        catchBlock.invokeInterfaceMethod(MethodDescriptors.LOCK_UNLOCK, lock);
+        catchBlock.throwException(catchBlock.getCaughtException());
+    }
+
+    private void implementRemove(ClassCreator contextInstances, List<BeanInfo> applicationScopedBeans,
+            Map<String, FieldDescriptor> idToField, FieldDescriptor lockField) {
+        MethodCreator remove = contextInstances
+                .getMethodCreator("remove", ContextInstanceHandle.class, String.class)
+                .setModifiers(ACC_PUBLIC);
+
+        StringSwitch strSwitch = remove.stringSwitch(remove.getMethodParam(0));
+        strSwitch.fallThrough();
```

Comment:
Any reason why you enable fallthrough here? Especially since all cases end with a `return`, so there cannot be any fallthrough.

Is that because the bytecode (or its decompiled form) looks better?

Reply:
Also, `computeIfAbsent` seems to enable fallthrough as well, while `getIfPresent` does not. Weird?

Reply:
> Is that because the bytecode (or its decompiled form) looks better?

That is a good question. So I did some experiments and the generated bytecode was cleaner and smaller. But I did not look into details.

It seems that our `Switch` implementation always attempts to break the scope at the end of the case block. However, the generated bytecode contains some garbage if the case blocks returns a value.

The garbage looks like:
```
100: nop
101: nop
102: athrow
```
And the StackMapTable contains more garbage:
```
rame_type = 255 /* full_frame */
          offset_delta = 5
          locals = []
          stack = [ class java/lang/Throwable ]
        frame_type = 255 /* full_frame */
          offset_delta = 2
          locals = [ class io/quarkus/arc/generator/Default_jakarta_enterprise_context_ApplicationScoped_ContextInstances, class java/lang/String, class java/util/function/Supplier, int ]
          stack = []
        frame_type = 255 /* full_frame */
          offset_delta = 5
          locals = []
          stack = [ class java/lang/Throwable ]
        frame_type = 255 /* full_frame */
          offset_delta = 2
          locals = [ class io/quarkus/arc/generator/Default_jakarta_enterprise_context_ApplicationScoped_ContextInstances, class java/lang/String, class java/util/function/Supplier, int ]
          stack = []
        frame_type = 255 /* full_frame */
          offset_delta = 5
          locals = []
          stack = [ class java/lang/Throwable ]
        frame_type = 255 /* full_frame */
          offset_delta = 2
          locals = [ class io/quarkus/arc/generator/Default_jakarta_enterprise_context_ApplicationScoped_ContextInstances, class java/lang/String, class java/util/function/Supplier, int ]
          stack = []
```

I'm not sure if we could easily detect the `goto` operation at the end of the case block and skip https://github.com/quarkusio/gizmo/blob/main/src/main/java/io/quarkus/gizmo/StringSwitchImpl.java#L56?

Reply:
OK, that's fair. Thanks!